compose.staging.yml

```diff
@@ -55,7 +46,7 @@ services:
     environment:
       - APP_ENV
       - CADDYPATH=/etc/caddycerts
-      - PROXY_HOST=web-host
+      - PROXY_HOST=web
       - PROXY_PORT=8000
       - HOST_NAME=staging.nepalmap.org # Change to staging server for nepalmap.org
       - TLS_EMAIL=test@example.com     # Enter a email address here link to LetsEncrypt certs
```

Comment:
When is the `TLS_EMAIL` variable used?

Is there value in some of these being set via environment variables?

Reply:
`TLS_EMAIL` is passed on to Caddy ( https://github.com/Code4Nepal/nepalmap_app/blob/dev/compose/caddy/Caddyfile#L3 ). It has to be set if you want to get the certificates non-interactively. https://caddyserver.com/docs/tls

Of course, you can hardcode them in the Caddyfile. But, this way you can override email addresses easily for different environments. The main idea is to have all the configuration injected. ( following ideas from https://12factor.net/config ) It's a matter of opinion really, so feel free to change it if you see fit.

If you don't want to list email, what we can do is pass through the env var like we do for `APP_ENV` which has to be set outside the compose file.